docs/_api/Tunix_API_DOCUMENTATION.md

```diff
@@ -0,0 +1,897 @@
+
+#### Data Flow
+
+Data flows through the system via structured dataclasses (e.g., `TrainingInput`, `TrainExample`). A typical flow is:
+1.  Raw data (text, preference pairs) is loaded.
+2.  A data processing pipeline (e.g., `tunix.models.gemma.data` using `grain`, or `process_dpo_record`) tokenizes and formats the data into a `TrainingInput` object.
+3.  The `Trainer`'s `_prepare_inputs` method further processes this input for the specific training algorithm (e.g., computing reference model log-probabilities in DPO).
+4.  The sharded data is fed into the JIT-compiled training step, which computes the loss and updates the model weights.
+
+#### Conclusion
+
+The Tunix library exhibits a mature and robust software architecture. By emphasizing modularity, performance, and a clear separation of concerns, it provides a powerful and flexible platform for both research and production-scale LLM fine-tuning. Its standout features are the scalable, disaggregated `RLCluster` for advanced RLHF and the pluggable inference architecture that supports both a native JAX sampler and the production-grade vLLM engine.
+
+## API Reference
+
+### SFT (Supervised Fine-Tuning)
+
+#### PEFT Trainer
+| Class/Function | Description |
+|---|---|
+| `PeftTrainer(...)` | PEFT trainer for LoRA. Only LoRA parameters are updated. |
+| `TrainingConfig()` | Configuration for the trainer. |
+| `TrainingInput()` | Dataclass representing the input for a single training step. |
+| `is_lora_enabled(model)` | Checks if the model has LoRA layers enabled. |
+| `time_measure([context])` | A context manager for measuring the execution time of a code block. |
+
+---
+
+##### `PeftTrainer`
+*class* `tunix.sft.peft_trainer.PeftTrainer`(*model: nnx.Module*, *optimizer: optax.GradientTransformation*, *training_config: TrainingConfig*)
+
+PEFT trainer for LoRA. Only LoRA parameters are updated.
+
+**Attributes:**
+- **model**: The model to train.
+- **config**: The training config.
+- **optimizer**: The optimizer to use. To monitor the learning rate at each step, use `optax.inject_hyperparams` to inject learning rate as a hyperparameter. For example: `optimizer = optax.inject_hyperparams(optax.sgd)(learning_rate=learning_rate_schedule)`
+- **loss_fn**: The loss function to use.
+- **eval_loss_fn**: The loss function to use for evaluation.
+- **gen_model_input_fn**: The function to generate model input from training input.
+- **checkpoint_manager**: The checkpoint manager to use.
+- **metrics_logger**: The metrics logger to use.
+- **is_managed_externally**: Whether the trainer is managed externally.
+- **training_hooks**: The training hooks to use.
+- **data_hooks**: The data hooks to use.
+
+**Methods:**
+
+`with_training_hooks(training_hooks: hooks.TrainingHooks) -> PeftTrainer`
+:   Sets the training hooks for the trainer.
+  - **Parameters:**
+    - `training_hooks`: An object implementing the `TrainingHooks` protocol.
+  - **Returns:** The trainer instance with the hooks configured.
```

Comment:
Whitespace matters: using 2 spaces instead of 4 makes list items render as a list:

<img width="1920" height="1080" alt="Screenshot from 2025-09-03 18-47-48" src="https://github.com/user-attachments/assets/94ff31fd-d0e5-46a5-a034-66dd355f8e0f" />